RIOT-OS/RIOT
# sys/suit: initial support for SUIT firmware updates

### Contribution description

This PR contains initial support for updating RIOT using an implementation of [SUIT draft version 4](https://tools.ietf.org/html/draft-moran-suit-manifest-04).

This currently only supports a limited use case: one component, two slots (one active one passive). Basically, it allows SUIT compliant infrastructure and tooling to update a RIOT node.

**Note**:  The implementation should not yet be considered production ready.

The implementation is based on a draft spec that will change in the near future. The implementation has also not yet gone through any security audit.
This should still be usable for less security sensitive deployments, and by getting this into master, we hope to get more exposure and feedback going towards production readiness.

The RIOT-fp project commits to maintaining and improving this implementation.

The PR is grouped in three commits:

- one commit adding the sys/suit module, including manifest handling and a CoAP based update service
- one commit adding necessary tooling (key- and manifest-generation) to dist/tools/suit
- one commit adding ```examples/suit_update```, show-casing SUIT usage in addition to providing an automated test case.

The commit history looks quite large due to the large list of dependencies. This PR should serve as tracking PR as long as there are open dependencies.

### Testing procedure

The PR ships with an example application in ```examples/suit_update```, which includes testing instructions using an automated test, which tests updating each slot over ethos.
The test is also run by the CI.

Apart from verifying that updating works for a given node, it would be very helpful to get feedback on the documentation and whether it describes the necessary steps in your setup.

### Issues/PRs references

Depends on ~~#11801~~, ~~#11802~~, ~~#11803~~, ~~#11805~~, ~~#11816~~, ~~#11707~~, ~~#11690~~, ~~#11697~~, and ~~[riotdocker #73](https://github.com/RIOT-OS/riotdocker/pull/73)~~
